addons/mail/controllers/discuss/channel.py

```diff
@@ -4,30 +4,55 @@
 
 from odoo import http
 from odoo.http import request
+from odoo.tools.misc import OrderedSet
 from odoo.addons.mail.controllers.webclient import WebclientController
 from odoo.addons.mail.models.discuss.mail_guest import add_guest_to_context
 from odoo.addons.mail.tools.discuss import Store
 
 
 class DiscussChannelWebclientController(WebclientController):
     """Override to add discuss channel specific features."""
-    def _process_request_for_all(self, store, **kwargs):
-        """Override to return channel as member and last messages."""
-        super()._process_request_for_all(store, **kwargs)
-        if kwargs.get("channels_as_member"):
-            channels = request.env["discuss.channel"]._get_channels_as_member()
+
+    # aggregate of channels to return, to batch them in a single query when all the fetch params have been processed
+    channels = None
+    add_channels_last_message = False
```

Comment:
This will not work with multiple registries.

Reply:
It felt wrong, but I couldn't remember why. Updated PR to use context, thank you for catching it!